linker/symbols_test.go

```diff
@@ -217,3 +206,19 @@ func TestSymbolExtensions(t *testing.T) {
 	assert.Equal(t, 1, len(extNums))
 	assert.Contains(t, extNums, protoreflect.FieldNumber(10101))
 }
+
+func parseAndLink(t *testing.T, contents string) Result {
+	h := reporter.NewHandler(nil)
+	fileAst, err := parser.Parse("test.proto", strings.NewReader(contents), h)
+	require.NoError(t, err)
+	parseResult, err := parser.ResultFromAST(fileAst, true, h)
+	require.NoError(t, err)
+	dep, err := protoregistry.GlobalFiles.FindFileByPath("google/protobuf/descriptor.proto")
+	require.NoError(t, err)
+	depAsFile, err := NewFileRecursive(dep)
+	require.NoError(t, err)
+	depFiles := Files{depAsFile}
+	linkResult, err := Link(parseResult, depFiles, nil, h)
+	require.NoError(t, err)
+	return linkResult
+}
```

Comment:
This is kinda gnarly -- no compiler means we have to hard-code things like the `google/protobuf/descriptor.proto` file. (The alternative would be putting way too many smarts into this test that really belong in the compiler and not here...)

WDYT?

Reply:
I prefer removing the potential cycle with imports personally. I think tests that require the compiler should be moved to the parent package.